schedd.go

```diff
@@ -78,12 +78,50 @@ func (s *Schedd) Address() string {
 // Query queries the schedd for job advertisements
 // constraint is a ClassAd constraint expression (use "true" to get all jobs)
 // projection is a list of attributes to return (use nil to get all attributes)
+// Deprecated: Use QueryWithOptions for pagination and default limits/projections
 func (s *Schedd) Query(ctx context.Context, constraint string, projection []string) ([]*classad.ClassAd, error) {
-	return s.queryWithAuth(ctx, constraint, projection, false)
+	return s.queryWithAuth(ctx, constraint, projection, false, nil)
 }
 
-// queryWithAuth performs the actual query with optional authentication
-func (s *Schedd) queryWithAuth(ctx context.Context, constraint string, projection []string, useAuth bool) ([]*classad.ClassAd, error) {
+// QueryWithOptions queries the schedd for job advertisements with pagination and limits
+// opts specifies query options including limit, projection, and pagination
+// Returns the matching job ads and pagination info
+func (s *Schedd) QueryWithOptions(ctx context.Context, constraint string, opts *QueryOptions) ([]*classad.ClassAd, *PageInfo, error) {
+	// Apply defaults if opts is nil
+	if opts == nil {
+		opts = &QueryOptions{}
+	}
+	effectiveOpts := opts.ApplyDefaults()
+
+	// Get effective projection
+	projection := effectiveOpts.GetEffectiveProjection(DefaultJobProjection())
+
+	// Query with the effective options
+	jobAds, err := s.queryWithAuth(ctx, constraint, projection, false, &effectiveOpts)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Build pagination info
+	pageInfo := &PageInfo{
```

Comment:
Have the page token be a base64-encoded version of `Cluster.Proc` as strings (so, if the last job returned is `14.1`, use `"14.1"` as the input to the base64).  This is the input to `queryAdsInternal` for the next round.  Assume if we got the limit or more, there are potentially more jobs and the page token should be returned.